vividus/src/main/java/org/vividus/runner/DeprecatedPropertiesReplacer.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.runner;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Properties;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+import org.apache.commons.io.FileUtils;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.ResourcePatternResolver;
+import org.vividus.configuration.BeanFactory;
+import org.vividus.configuration.DeprecatedPropertiesHandler;
+import org.vividus.configuration.Vividus;
+
+public class DeprecatedPropertiesReplacer
+{
+    private static final String PLACEHOLDER_PREFIX = "${";
+    private static final String PLACEHOLDER_SUFFIX = "}";
+    private static final String DEPRECATED_PROPERTIES_CLASSPATH = "classpath*:/properties/**/deprecated.properties";
+
+    private static final Properties DEPRECATED_PROPERTIES = new Properties();
+
+    private final ResourcePatternResolver resourcePatternResolver;
+
+    public DeprecatedPropertiesReplacer(ResourcePatternResolver resourcePatternResolver)
+    {
+        this.resourcePatternResolver = resourcePatternResolver;
+    }
+
+    public static void main(String[] args) throws IOException, ParseException
+    {
+        Vividus.init();
+
+        Options options = new Options();
+        Option directoryOption = new Option("rl", "resourceLocation", true, "Directory to find properties.");
+        options.addOption(directoryOption);
```

Comment:
hmmm why not to travers everything inside properties directory + configuration.propertieis entry point?